src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -3638,6 +3642,29 @@ private ConstantValue GetConstantValueOrThrow(ConstantHandle handle)
 
             matchedName = null;
             return (FirstIndex: -1, SecondIndex: -1);
+
+            void ensureCaseInsensitiveDictionary()
+            {
+                if (_lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap != null)
+                {
+                    return;
+                }
+
+                if (_lazyForwardedTypesToAssemblyIndexMap.Count == 0)
+                {
+                    _lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap = s_sharedEmptyCaseInsensitiveForwardedTypes;
+                    return;
+                }
+
+                var caseInsensitiveMap = new Dictionary<string, (int FirstIndex, int SecondIndex)>(StringComparer.OrdinalIgnoreCase);
+
+                foreach (var (key, indexPair) in _lazyForwardedTypesToAssemblyIndexMap)
```

Comment:
>_lazyForwardedTypesToAssemblyIndexMap [](http://example.com/codeflow?start=49&length=37)

I assume ordering doesn't matter in the case of matching keys?

Reply:
Correct. The existing code preferred whatever came first in the dictionary, which is pretty much undefined since it isn't an ordered dictionary in any way.